pinot-core/src/main/java/org/apache/pinot/core/operator/combine/BaseCombineOperator.java

```diff
@@ -180,4 +181,20 @@ public List<Operator> getChildOperators() {
    * Invoked when {@link #processSegments()} is finished (called in the finally block).
    */
   protected abstract void onProcessSegmentsFinish();
+
+  protected static void onOperatorException(Operator operator, Exception e) {
```

Comment:
Let's call it `wrapOperatorException()` and return `RuntimeException` so that we can directly throw it out in the caller. Currently it is a little bit hard to read because exception can be thrown in 2 places